Assets/MixedRealityToolkit/Extensions/BoundsExtensions.cs

```diff
@@ -354,54 +354,62 @@ public static void GetColliderBoundsPoints(GameObject target, List<Vector3> boun
+            }
+        }
 
-                if(colliders[i] is SphereCollider)
-                {
-                    SphereCollider sc = colliders[i] as SphereCollider;
-                    Bounds sphereBounds = new Bounds(sc.center, Vector3.one * sc.radius * 2);
-                    sphereBounds.GetFacePositions(sc.transform, ref corners);
-                    boundsPoints.AddRange(corners);
-                }
-                else if(colliders[i] is BoxCollider)
-                {
-                    BoxCollider bc = colliders[i] as BoxCollider;
-                    Bounds boxBounds = new Bounds(bc.center, bc.size);
-                    boxBounds.GetCornerPositions(bc.transform, ref corners);
-                    boundsPoints.AddRange(corners);
+        /// <summary>
+        /// Method to get bounds from a single Collider
+        /// </summary>
+        /// <param name="collider">Target collider</param>
+        /// <param name="boundsPoints">array reference that gets filled with points</param>
+        /// <param name="ignoreLayers">layerMask to simplify search</param>
+        public static void GetColliderBoundsPoints(Collider collider, List<Vector3> boundsPoints, LayerMask ignoreLayers)
+        {
+            if (ignoreLayers == (1 << collider.gameObject.layer | ignoreLayers)) { return; }
 
-                }
-                else if(colliders[i] is MeshCollider)
+            if (collider is SphereCollider)
+            {
+                SphereCollider sc = collider as SphereCollider;
+                Bounds sphereBounds = new Bounds(sc.center, Vector3.one * sc.radius * 2);
+                sphereBounds.GetFacePositions(sc.transform, ref corners);
+                boundsPoints.AddRange(corners);
+            }
+            else if (collider is BoxCollider)
+            {
+                BoxCollider bc = collider as BoxCollider;
+                Bounds boxBounds = new Bounds(bc.center, bc.size);
+                boxBounds.GetCornerPositions(bc.transform, ref corners);
+                boundsPoints.AddRange(corners);
+
+            }
+            else if (collider is MeshCollider)
+            {
+                MeshCollider mc = collider as MeshCollider;
+                Bounds meshBounds = mc.sharedMesh.bounds;
+                meshBounds.GetCornerPositions(mc.transform, ref corners);
+                boundsPoints.AddRange(corners);
+            }
+            else if (collider is CapsuleCollider)
+            {
+                CapsuleCollider cc = collider as CapsuleCollider;
+                Bounds capsuleBounds = new Bounds(cc.center, Vector3.zero);
+                switch (cc.direction)
                 {
-                    MeshCollider mc = colliders[i] as MeshCollider;
-                    Bounds meshBounds = mc.sharedMesh.bounds;
-                    meshBounds.GetCornerPositions(mc.transform, ref corners);
-                    boundsPoints.AddRange(corners);
+                    case 0:
```

Comment:
can you make the numeric values private const ints at top or at the very least add a comment referring to which direction each value corresponds to?
https://docs.unity3d.com/ScriptReference/CapsuleCollider-direction.html

Really surprised Unity used regular ints instead of an enum